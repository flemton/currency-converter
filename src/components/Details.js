/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable react/prop-types */
import React, { useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import getConverted from './getConverted';
import { clearConverted } from '../redux/stocks/currenciesSlice';

const Details = () => {
  const converted = useSelector((state) => state.currencies);
  const dispatch = useDispatch();
  const [from, setFrom] = useState(converted.currency || '--');
  const [to, setTo] = useState('usd');

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(clearConverted());
    dispatch(getConverted({ from: e.target.from.value, to: e.target.to.value }));
  };

  useEffect(() => {
    dispatch(clearConverted());
    dispatch(getConverted({ from, to: 'usd' }));
  }, [1]);

  const convertedA = converted.converted ? Object.entries(converted.converted) : [{ one: 'one', none: 'Check currencies' }, { from: '--', to: '--' }];
  return (
    <div>
      <div>
        <form onSubmit={(e) => handleSubmit(e)}>
          <label>
            From
            <input required type="text" id="from" name="from" onBlur={(e) => setFrom(e.target.value)} />
          </label>
          <label>
            To
            <input required type="text" id="to" name="to" onBlur={(e) => setTo(e.target.value)} />
          </label>
          <button type="submit">Convert</button>
        </form>
      </div>
      <h4>
        Date:
        {' '}
        {convertedA[0][1]}
      </h4>
      <p>
        1
        {' '}
        {from.toUpperCase()}
        {' equals '}
        {convertedA[1][1]}
        {' '}
        {to.toUpperCase()}
      </p>
    </div>
  );
};

export default Details;
