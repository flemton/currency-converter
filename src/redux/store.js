import { configureStore } from '@reduxjs/toolkit';
import currenciesReducer from './stocks/currenciesSlice';

const store = configureStore({
  reducer: {
    currencies: currenciesReducer,
  },
});

export default store;
