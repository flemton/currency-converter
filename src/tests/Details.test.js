import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import React from 'react';
import Details from '../components/Details';
import store from '../redux/store';

describe('details', () => {
  test('test if details renders', () => {
    const tree = render(
      <Provider store={store}>
        <Details />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
