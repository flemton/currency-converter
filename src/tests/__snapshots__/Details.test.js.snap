// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`det test if home renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details"
      >
        Convert
        <div>
          <form
            class="form"
          >
            <label>
              <input
                id="from"
                name="from"
                placeholder="from"
                required=""
                type="text"
              />
            </label>
            <label>
              <input
                id="to"
                name="to"
                placeholder="to"
                required=""
                type="text"
              />
            </label>
            <button
              type="submit"
            >
              GO!
            </button>
          </form>
        </div>
        <h4>
          Date:
           
          Check currencies
        </h4>
        <p>
          1
           
          --
           equals 
          --
           
          USD
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="details"
    >
      Convert
      <div>
        <form
          class="form"
        >
          <label>
            <input
              id="from"
              name="from"
              placeholder="from"
              required=""
              type="text"
            />
          </label>
          <label>
            <input
              id="to"
              name="to"
              placeholder="to"
              required=""
              type="text"
            />
          </label>
          <button
            type="submit"
          >
            GO!
          </button>
        </form>
      </div>
      <h4>
        Date:
         
        Check currencies
      </h4>
      <p>
        1
         
        --
         equals 
        --
         
        USD
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`detail test if details renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="details"
      >
        Convert
        <div>
          <form
            class="form"
          >
            <label>
              <input
                id="from"
                name="from"
                placeholder="from"
                required=""
                type="text"
              />
            </label>
            <label>
              <input
                id="to"
                name="to"
                placeholder="to"
                required=""
                type="text"
              />
            </label>
            <button
              type="submit"
            >
              GO!
            </button>
          </form>
        </div>
        <h4>
          Date:
           
          Check currencies
        </h4>
        <p>
          1
           
          --
           equals 
          --
           
          USD
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="details"
    >
      Convert
      <div>
        <form
          class="form"
        >
          <label>
            <input
              id="from"
              name="from"
              placeholder="from"
              required=""
              type="text"
            />
          </label>
          <label>
            <input
              id="to"
              name="to"
              placeholder="to"
              required=""
              type="text"
            />
          </label>
          <button
            type="submit"
          >
            GO!
          </button>
        </form>
      </div>
      <h4>
        Date:
         
        Check currencies
      </h4>
      <p>
        1
         
        --
         equals 
        --
         
        USD
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`home test if home renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home"
        data-testid="home"
      >
        <div>
          Loading...
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home"
      data-testid="home"
    >
      <div>
        Loading...
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
